spec.bs

```diff
@@ -777,6 +780,82 @@ To <dfn>generate and score bids</dfn> given an [=auction config=] |auctionConfig
 
 </div>
 
+<div algorithm>
+To <dfn>build an interest group passed to generateBid</dfn> given an [=interest group=] |ig|:
+  
+  1. Let |igGenerateBid| be a new {{GenerateBidInterestGroup}}.

```

Comment:
Can you store the {{GenerateBidInterestGroup}} as a field of the [=interest group=] to avoid needing to duplicate the conversion logic between here and https://pr-preview.s3.amazonaws.com/qingxinwu/turtledove/pull/601.html#dom-navigator-joinadinterestgroup?

Reply:
Paul had a great point that the {{GenerateBidInterestGroup}} also needs to be updated during IG updates, which adds a complication of having to update both [=interest group=] and its {{GenerateBidInterestGroup}}.

Reply:
Which fields would need to be duplicated? I imagine that you'd remove fields like https://pr-preview.s3.amazonaws.com/qingxinwu/turtledove/pull/601.html#interest-group-owner and https://pr-preview.s3.amazonaws.com/qingxinwu/turtledove/pull/601.html#interest-group-name when adding the {{GenerateBidInterestGroup}}, but you'd have to keep things like https://pr-preview.s3.amazonaws.com/qingxinwu/turtledove/pull/601.html#interest-group-joining-origin that don't get exposed down to the bidding JS.

Reply:
Yes, and for some fields that'll argue for duplicating the field, if their parsed form is used by the algorithms here in addition to passing the unparsed form to the script-runner scripts.

You also need to continue to parse-and-serialize the fields when converting from the `joinInterestGroup()` argument to the script-runner argument dictionary, to avoid letting callers smuggle extra information in extraneous parts of the string.

Reply:
Now I'm thinking whether that'll be worth doing so, rather than just duplicating the conversion logic at one place which seems easier to me.
Some fields (e.g., owner) are used many times in the spec, in the parsed (e.g., [=origin=]) form, so having both the origin and USVString owner in [=interest group=] seems to be needed. But, if we duplicate some fields, then we need to handle setting and updating both (in different types, like owner in both origin and USVString). And I think browser's [=interest group set=] should not store the same field as two different fields, so have to add a note explaining the IDL field is not stored, but just to be passed to javascript.

Reply:
It might not be worth it if you need to duplicate lots of fields, but just storing the owner separately is probably simple enough. Sometimes when writing specs, just like writing code, you have to try a refactoring before you can tell for sure if it'll pay off. I expect it to look simpler to store the owner in two places up front, rather than defining an IDL field but only filing it in as part of running script, but I could be wrong.

Reply:
Sounds good!